samples/python/tutorial_code/dnn/dnn_conversion/common/accuracy_eval.py

```diff
@@ -0,0 +1,150 @@
+import os
+import sys
+import time
+from abc import ABCMeta, abstractmethod
+
+import cv2
+import numpy as np
+
+
+# https://github.com/opencv/opencv/blob/master/modules/dnn/test/imagenet_cls_test_alexnet.py#L98
+def get_correct_answers(img_list, img_classes, net_output_blob):
+    correct_answers = 0
+    for i in range(len(img_list)):
+        indexes = np.argsort(net_output_blob[i])[-5:]
+        correct_index = img_classes[img_list[i]]
+        if correct_index in indexes:
+            correct_answers += 1
+    return correct_answers
+
+
+# https://github.com/opencv/opencv/blob/master/modules/dnn/test/imagenet_cls_test_alexnet.py#L26
+class DataFetch(object):
+    imgs_dir = ''
+    frame_size = 0
+    bgr_to_rgb = False
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def preprocess(self, img):
+        pass
+
+    def get_batch(self, imgs_names):
+        assert type(imgs_names) is list
+        batch = np.zeros((len(imgs_names), 3, self.frame_size, self.frame_size)).astype(np.float32)
+        for i in range(len(imgs_names)):
+            img_name = imgs_names[i]
+            # changed to os.path.join
+            img_file = os.path.join(self.imgs_dir, img_name)
+            assert os.path.exists(img_file)
+            img = cv2.imread(img_file, cv2.IMREAD_COLOR)
+            min_dim = min(img.shape[-3], img.shape[-2])
+            resize_ratio = self.frame_size / float(min_dim)
+            img = cv2.resize(img, (0, 0), fx=resize_ratio, fy=resize_ratio)
+            cols = img.shape[1]
+            rows = img.shape[0]
+            y1 = round((rows - self.frame_size) / 2)
+            y2 = round(y1 + self.frame_size)
+            x1 = round((cols - self.frame_size) / 2)
+            x2 = round(x1 + self.frame_size)
+            img = img[y1:y2, x1:x2]
+            if self.bgr_to_rgb:
+                img = img[..., ::-1]
+            image_data = img[:, :, 0:3].transpose(2, 0, 1)
```

Comment:
I suggest to split this code to several small functions like `center_crop` and `preprocess_image` to make it more readable.